Add rulset metadata attribute for structure tree node label

thomaslow opened the pull request:
Fixes #5882 

This pull request allows to declare what metadata keys are used as node labels in the structure tree (`use="structureTreeTitle"`).

- A user may choose between the options `Type`, `Title` and `Type - Title` in the metadata editor
- A warning is shown in case the `ruleset.xml` does not contain at least one metadata key with `use="structureTreeTitle"`
- The old configuration option `metsEditor.titleMetadata` was removed from `kitodo_config.properties`

Example declaration in `rulset.xml`:

```xml
<ruleset xmlns="http://names.kitodo.org/ruleset/v2" lang="en">
    <declaration>
        ...
        <key id="TitleDocMain" use="structureTreeTitle">
            <label>main title</label>
            <label lang="de">Haupttitel</label>
        </key>
        ...
    </declaration>
</ruleset>
```

Demo

https://github.com/user-attachments/assets/cac83ae2-7dfd-4d24-bfdb-165015248ae0

henning-gerhardt commented:
@thomaslow : can use please extend the `ruleset.xsd` file for this new entry? Without this it is not possible anymore to validate the rulesets which use this new attribute value.

thomaslow commented:
@henning-gerhardt Yes, of course. I will adapt the `ruleset.xsd` and add unit/integration tests later. This pull request is still work-in-progress. I stopped halfway through the implementation, because I felt that the current solution is not ideal. My concerns and questions are mentioned in #5882.

thomaslow commented:
@henning-gerhardt I updated both `ruleset.xsd`, added the new option to the `simple-book.xml` example ruleset, and added or updated unit/integration/selenium tests.

solth commented on Kitodo/src/main/java/org/kitodo/production/services/dataeditor/DataEditorService.java:
```diff
@@ -143,13 +131,14 @@ public static String getTitleValue(LogicalDivision element, String metadataTitle
                     .collect(Collectors.toList());
         }
-        Optional<String> metadataTitle = metadata.stream()
+        String metadataTitle = metadata.stream()
```
Could the result of this method call not be returned directly? I am not sure if storing it to a separate variable for just one line has any benifit.

thomaslow replied:
Yes, done.

solth commented on Kitodo/src/main/webapp/WEB-INF/templates/includes/metadataEditor/logicalStructure.xhtml:
```diff
@@ -20,10 +20,12 @@
+            <f:selectItem itemLabel="#{msgs.typ}" itemValue="type" />
+            <f:selectItem itemLabel="#{msgs.title}" itemValue="title" />
+            <f:selectItem itemLabel="#{msgs.typ} - #{msgs.title}" itemValue="type+title" />
+            <!--<f:selectItem itemLabel="#{msgs.typ} - #{msgs.count}" itemValue="type+count" />-->
```
Do you plan to implement this option as well? If not I would suggest to remove the comment. If we ever come around to adding the `count` option this should be (re-)added, in my opinion.

thomaslow replied:
Sorry. Forget to remove the comments that I left to remind me about the count option. I removed it.

solth commented on Kitodo/src/main/webapp/WEB-INF/templates/includes/metadataEditor/logicalStructure.xhtml:
```diff
@@ -125,9 +127,13 @@
                 <h:outputText value="#{DataEditorForm.getStructureElementTitle(logicalNode.dataObject)}"
-                              rendered="#{DataEditorForm.structurePanel.titleMetadata ne 'type'}"/>
+                              rendered="#{DataEditorForm.structurePanel.nodeLabelOption.contains('title') and !logicalNode.linked}"/>
+                <!--<h:outputText value="1234"
```
Does this comment have any particular purpose?

thomaslow replied:
No, sorry. I removed it.

solth commented on Kitodo/src/test/java/org/kitodo/production/forms/dataeditor/DataEditorFormTest.java:
```diff
@@ -73,16 +75,15 @@ public void shouldCheckIfMediaIsSelected() {
-        DataEditorForm dataEditorForm = new DataEditorForm();
-        dataEditorForm.getStructurePanel().setTitleMetadata(TITLE_METADATA_KEY);
+        // define which metadata keys to use as title
+        Collection<String> metadataKeys = Arrays.asList(TITLE_METADATA_KEY);
```
```suggestion
        Collection<String> metadataKeys = List.of(TITLE_METADATA_KEY);
```
This can be simplified using `List.of`for lists containing only one element, which - according to my IDE - saves some memory.

thomaslow replied:
Done

solth commented on Kitodo/src/test/java/org/kitodo/production/forms/dataeditor/DataEditorFormTest.java:
```diff
@@ -15,6 +15,8 @@
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
+import java.util.Arrays;
```
```suggestion
```

thomaslow replied:
Done

solth reviewed (approved):
Works very well and code looks good. Just have some very small suggestions that could perhaps be incorporated.

solth commented:
@thomaslow also please rebase against current master to resolve code conflicts that were introduced by merging #6255

thomaslow commented on Kitodo/src/main/resources/kitodo_config.properties:
```diff
@@ -289,9 +289,6 @@ metsEditor.displayFileManipulation=true
 metsEditor.pageSeparators=" "
 
-# Priority list of metadata keys used to display title information in the metadata editors structure and gallery panels
-metsEditor.titleMetadata=TitleDocMain
```
Done

thomaslow commented:
@BartChris I added this text to the Ruleset wiki page:

`use="structureTreeTitle" Das Feld wird im Strukturbaum des Metadateneditor als Anzeigetext für die Strukturelemente verwendet, falls vom Nutzer die Anzeigeoption "Titel" ausgewählt wurde.`

See [wiki section](https://github.com/kitodo/kitodo-production/wiki/Regelsatz-3.x/#schl%C3%BCssel-mit-besonderer-verwendung).
